libp2p/NodeTable.cpp

```diff
@@ -557,19 +562,31 @@ void NodeTable::doCheckEvictions()
         
         bool evictionsRemain = false;
         list<shared_ptr<NodeEntry>> drop;
+        list<shared_ptr<NodeEntry>> active;
+
         {
             Guard le(x_evictions);
             Guard ln(x_nodes);
             for (auto& e: m_evictions)
                 if (chrono::steady_clock::now() - e.second.evictedTimePoint > c_reqTimeout)
-                    if (m_nodes.count(e.second.newNodeID))
-                        drop.push_back(m_nodes[e.second.newNodeID]);
+                    if (m_nodes.count(e.first))
+                    {
+                        drop.push_back(m_nodes[e.first]);
+                        if (m_nodes.count(e.second.newNodeID))
```

Comment:
Sorry I've merged my changes before you, you'll need to rebase and to resolve conflicts (should be not hard, my changes were trivial so far)
Around here I changed `m_nodes.count` call to `m_nodes.find`, please do the same with this new `count` call